x-pack/plugins/security_solution/public/entity_analytics/components/asset_criticality/use_asset_criticality.test.ts

```diff
@@ -35,30 +42,82 @@ describe('useAssetCriticality', () => {
-  it('does not call privileges API when hasEntityAnalyticsCapability is false', async () => {
-    mockUseHasSecurityCapability.mockReturnValue(false);
-    mockUseUiSettings.mockReturnValue([true]);
+  describe('useAssetCriticalityPrivileges', () => {
+    it('does not call privileges API when hasEntityAnalyticsCapability is false', async () => {
+      mockUseHasSecurityCapability.mockReturnValue(false);
+      mockUseUiSettings.mockReturnValue([true]);
 
-    await renderQuery(() => useAssetCriticalityPrivileges('test_entity_name'), 'isSuccess');
+      await renderQuery(() => useAssetCriticalityPrivileges('test_entity_name'), 'isSuccess');
 
-    expect(mockFetchAssetCriticalityPrivileges).not.toHaveBeenCalled();
-  });
+      expect(mockFetchAssetCriticalityPrivileges).not.toHaveBeenCalled();
+    });
+
+    it('calls privileges API when hasEntityAnalyticsCapability and UiSettings are enabled', async () => {
+      mockUseHasSecurityCapability.mockReturnValue(true);
+      mockUseUiSettings.mockReturnValue([true]);
+
+      await renderQuery(() => useAssetCriticalityPrivileges('test_entity_name'), 'isSuccess');
 
-  it('calls privileges API when hasEntityAnalyticsCapability and UiSettings are enabled', async () => {
-    mockUseHasSecurityCapability.mockReturnValue(true);
-    mockUseUiSettings.mockReturnValue([true]);
+      expect(mockFetchAssetCriticalityPrivileges).toHaveBeenCalled();
+    });
 
-    await renderQuery(() => useAssetCriticalityPrivileges('test_entity_name'), 'isSuccess');
+    it('does not call privileges API when UI Settings is disabled', async () => {
+      mockUseHasSecurityCapability.mockReturnValue(true);
+      mockUseUiSettings.mockReturnValue([false]);
 
-    expect(mockFetchAssetCriticalityPrivileges).toHaveBeenCalled();
+      await renderQuery(() => useAssetCriticalityPrivileges('test_entity_name'), 'isSuccess');
+
+      expect(mockFetchAssetCriticalityPrivileges).not.toHaveBeenCalled();
+    });
   });
 
-  it('does not call privileges API when UI Settings is disabled', async () => {
-    mockUseHasSecurityCapability.mockReturnValue(true);
-    mockUseUiSettings.mockReturnValue([false]);
+  describe('useAssetCriticalityData', () => {
+    it('calls delete api when the mutation is called with unassigned criticality level', async () => {
+      mockFetchAssetCriticalityPrivileges.mockResolvedValue({ has_all_required: true });
+      mockDeleteAssetCriticality.mockResolvedValue({});
+      mockCreateAssetCriticality.mockResolvedValue({});
+      const entity: Entity = { name: 'test_entity_name', type: 'host' };
+
+      const { mutation } = await renderQuery(
+        () => useAssetCriticalityData({ entity }),
+        'isSuccess'
+      );
+
+      await renderMutation(() =>
+        mutation.mutate({
+          idField: 'test_entity_type.name',
+          idValue: 'test_entity_name',
+          criticalityLevel: 'unassigned',
+        })
+      );
+
+      await Promise.resolve(); // why do I need this here?
```

Comment:
Does the test fail without this line?

If that is the case, it looks like your test needs to wait for all promises to be resolved. I have seen different ways to do that, like: 
```js
await new Promise(setImmediate);

await new Promise(process.nextTick);
```
Another option is wrap your assertion with `waitFor`, like:
```js
await waitFor(() => {
   expect(handleButtonSendMessage).toHaveBeenCalledWith(promptText);
});
```

In case you don't need to wait for a promise but for react to finish all updates, you could do:

```js
const { waitForNextUpdate } = renderHook(() => useMyHook());
await waitForNextUpdate();

```

Reply:
@machadoum Sorry I did manage to fix this if you look at the latest code, adding async to the renderMutation callback fixed it, I am not going to lie I have no idea why

Reply:
Here is the fix https://github.com/elastic/kibana/pull/180603/files#diff-cd66b2191127efc3418cf5d220b4236f0cb2ce373e5e5a1cf4876a063c7d072cR86